File: app/src/debug/java/com/handstandsam/shoppingapp/debug/DebugDispatcher.java
package com.handstandsam.shoppingapp.debug;


import android.content.Context;
import android.net.Uri;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

import okhttp3.mockwebserver.Dispatcher;
import okhttp3.mockwebserver.MockResponse;
import okhttp3.mockwebserver.RecordedRequest;

public class DebugDispatcher extends Dispatcher {

    Context context;

    public DebugDispatcher(Context context) {
        this.context = context;
    }

    @Override
    public MockResponse dispatch(RecordedRequest request) throws InterruptedException {
        Uri uri = Uri.parse(request.getPath());
        String username = uri.getLastPathSegment();

        String assetResponseJson = getAsset("user_" + username.toLowerCase() + ".json");
        if (assetResponseJson != null && !assetResponseJson.isEmpty()) {
            return new MockResponse().setBody(assetResponseJson).setResponseCode(200);
        } else {
            return new MockResponse().setResponseCode(404);
        }
    }

    private String getAsset(String filename) {
        StringBuilder total = new StringBuilder();
        try {
            InputStream inputStream = context.getAssets().open(filename);
            BufferedReader r = new BufferedReader(new InputStreamReader(inputStream));
            String line;
            while ((line = r.readLine()) != null) {
                total.append(line).append('\n');
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return total.toString();
    }
}

File: app/src/debug/java/com/handstandsam/shoppingapp/debug/DebugDispatcher.kt
package com.handstandsam.shoppingapp.debug


import android.content.Context
import android.net.Uri
import okhttp3.mockwebserver.Dispatcher
import okhttp3.mockwebserver.MockResponse
import okhttp3.mockwebserver.RecordedRequest
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

class DebugDispatcher(internal var context: Context) : Dispatcher() {

    @Throws(InterruptedException::class)
    override fun dispatch(request: RecordedRequest): MockResponse {
        val uri = Uri.parse(request.path)
        val username = uri.lastPathSegment

        val assetResponseJson = getAsset("user_" + username.toLowerCase() + ".json")
        return if (assetResponseJson != null && !assetResponseJson.isEmpty()) {
            MockResponse().setBody(assetResponseJson).setResponseCode(200)
        } else {
            MockResponse().setResponseCode(404)
        }
    }

    private fun getAsset(filename: String): String {
        val total = StringBuilder()
        try {
            val inputStream = context.assets.open(filename)
            val r = BufferedReader(InputStreamReader(inputStream))
            var line: String = r.readLine()
            while (line != null) {
                total.append(line).append('\n')
                line = r.readLine()
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return total.toString()
    }
}
